### PR TITLE
fix(isoformat): accept unicode_literals

### LINKED ISSUE #102: `jdatetime.datetime.isoformat` fails on python 2.7 when called after `from __future__ import unicode_literals`
```python
$ python2
Python 2.7.13 (default, Apr 16 2021, 14:02:03)
[GCC 6.3.0 20170516] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import jdatetime
>>> jdatetime.datetime(1398, 4, 11, 11, 6, 5, 123456).isoformat('T')
u'1398-04-11T11:06:05.123456'
>>> from __future__ import unicode_literals
>>> jdatetime.datetime(1398, 4, 11, 11, 6, 5, 123456).isoformat('T')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/nfs/labstore-secondary-tools-home/5j9/venv2/local/lib/python2.7/site-packages/jdatetime/__init__.py", line 1269, in isoformat
    'argument 1 must be a single character: {}'.format(sep)
AssertionError: argument 1 must be a single character: T
>>>

```

I believe the code above should work, even after the `__future__` import; similar to the way, for example, `jdatetime.datetime(1398, 4, 11, 11, 6, 5, 123456).strftime('%Y')` works.

# Please include your runtime information

* jdatetime version: 3.7.0
* Python version: 2.7
* OS: Linux

### PR DESCRIPTION
**Pull request by 5j9**
closes #102

### CODE REVIEW

File: jdatetime/__init__.py [hunk 1]
```diff
@@ -39,6 +39,7 @@
 
 if sys.version_info[0] >= 3:  # py3
     _int_types = (int,)
+    basestring = str
```
Comment (5j9) [inline jdatetime/__init__.py:42]: perhaps it's better to make this variable private? (`basestring` -> `_basestring`)

**Comment by 5j9** [comment]
Apparently flake8 does not know about `basestring` in python 2. I'll mark it as noqa.

**Comment by slashmili** [comment]
@5j9 Thanks for the contribution. Looks good to me. I'd say wait for @hramezani as well. I'm a bit rusty in Python!

BTW wondering if we should bump the MINOR or PACTH version for this. Asking since this has been broken since the beginning!

**Review by hramezani — APPROVED** [review]
Thanks @5j9 :+1:

**Comment by hramezani** [comment]
> BTW wondering if we should bump the MINOR or PACTH version for this. Asking since this has been broken since the beginning!

I would go for the MINOR

**Pull request merged**